src/components/date-time-input/date-time-input.ts

```diff
@@ -0,0 +1,652 @@
+import { html } from 'lit';
+import { property } from 'lit/decorators.js';
+import { live } from 'lit/directives/live.js';
+import {
+  DatePartDeltas,
+  DatePartInfo,
+  DateParts,
+  DatePart,
+  DateTimeUtil,
+} from './date-util';
+import { blazorTwoWayBind, watch } from '../common/decorators';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { IgcMaskInputBaseComponent } from '../mask-input/mask-input-base';
+import { partNameMap } from '../common/util';
+import { IgcInputEventMap } from '../input/input-base';
+import { EventEmitterMixin } from '../common/mixins/event-emitter';
+import { AbstractConstructor } from '../common/mixins/constructor';
+
+export interface IgcDateTimeInputEventMap
+  extends Omit<IgcInputEventMap, 'igcChange'> {
+  igcChange: CustomEvent<Date | null>;
+}
+
+export default class IgcDateTimeInputComponent extends EventEmitterMixin<
```

Comment:
Add [documentation](https://github.com/IgniteUI/igniteui-webcomponents/wiki/How-to-document-your-component)